src/lightning/fabric/wrappers.py

```diff
@@ -82,6 +86,21 @@ def step(self, closure: Optional[Callable] = None) -> Any:
                 hook(strategy=self._strategy, optimizer=optimizer)
         return output
 
+    def _refresh(self) -> None:
+        """Refreshes the ``__dict__`` so that it matches the internal states in the wrapped optimizer.
+
+        This is only needed to present the user with an updated view in case they inspect the state of this wrapper.
```

Comment:
Do you mean that the actual torch optimizer object would have the correct state? Because from my comment in #18482 it seems that both the wrapper and the actual optimizer did not have a state.